libs/designsystem/src/lib/components/button/button.component.ts

```diff
@@ -66,23 +65,32 @@ export class ButtonComponent implements AfterContentInit {
 
   @ContentChild(IconComponent, { static: false }) icon: IconComponent;
   @ContentChild(IconComponent, { static: false, read: ElementRef })
-  iconDomNode: ElementRef;
+  iconElementRef: ElementRef<HTMLElement>;
   private _hasSlottedContent = false;
 
   ngAfterContentInit(): void {
-    if (this.iconDomNode && this.iconDomNode.nativeElement) {
-      const prev = this.iconDomNode.nativeElement.previousSibling;
-      const next = this.iconDomNode.nativeElement.nextSibling;
-      if (prev && prev.nodeName !== '#comment') {
+    if (this.iconElementRef && this.iconElementRef.nativeElement) {
+      const iconElement = this.iconElementRef.nativeElement;
+      const prev = this.findNonCommentSibling(iconElement, 'previous');
+      const next = this.findNonCommentSibling(iconElement, 'next');
+      if (prev) {
         this._hasSlottedContent = true;
         this._isIconLeft = false;
         this._isIconRight = true;
       }
-      if (next && next.nodeName !== '#comment') {
+      if (next) {
         this._hasSlottedContent = true;
         this._isIconLeft = true;
         this._isIconRight = false;
       }
     }
   }
+
+  private findNonCommentSibling(element: HTMLElement, direction: 'next' | 'previous') {
+    const sibling = direction === 'previous' ? element.previousSibling : element.nextSibling;
+    if (sibling != null && sibling.nodeName === '#comment') {
+      return this.findNonCommentSibling(sibling as HTMLElement, direction);
```

Comment:
The cast can be removed by generalizing the paramater type: 
```suggestion
  private findNonCommentSibling(element: Element, direction: 'next' | 'previous') {
    const sibling = direction === 'previous' ? element.previousSibling : element.nextSibling;
    if (sibling != null && sibling.nodeName === '#comment') {
      return this.findNonCommentSibling(sibling, direction);
```